### PR TITLE
Added Card Widgets (_DevListCardFirst and _DevListCardSecond)

### PR DESCRIPTION
**Pull request by jose-bamboo**
File: dev_list.dart
Added card widgets to list of devs
Icons to be updated to socials, temporarily material icons
Can try out card dark mode by changing isDark to true

File: devboard_page.dart
modified DevboardPage widget, for fixed card list view sizing
Added ListType enum
Added other card types to Devboard Page

File: devs.json
Added socials

Card demo 1: https://imgur.com/vWuCXhX
Card demo 2: https://imgur.com/V1jslVt

### CODE REVIEW

**Comment by Jansalvador1445** [comment]
Woah! I liked the design!  But it has conflicts :( Please rebase.

**Comment by jose-bamboo** [comment]
This is noted. Thanks!

**Comment by Jansalvador1445** [comment]
Still has `Conflict`. I'm excited more to merge to! :D

**Comment by jose-bamboo** [comment]
Hello, could I ask for help on this. I'm very new to this as it is my first time contributing.

**Comment by Jansalvador1445** [comment]
Sure, wait let me, check your repo.

**Comment by jose-bamboo** [comment]
I think it's working, though I may have to comment out some of the list views in the devboard_page.dart

**Comment by Jansalvador1445** [comment]
Added changes here, but please fix the `overflow` of the text :D

**Comment by Jansalvador1445** [comment]
And the card shadow, i think i accidentally removed it.

**Comment by jose-bamboo** [comment]
Fixed the overflow issue by increasing the height of the card and added maxLines in Text widgets to avoid overflowing

**Comment by Jansalvador1445** [comment]
Will test it now.

File: lib/features/devboard/devboard_page.dart [hunk 1]
```diff
@@ -27,29 +29,18 @@ class _DevboardPageState extends State<DevboardPage> {
...
-                  ),
-                ),
-              ),
+        child: FutureProvider<List<Dev>>(
+          create: (_) => devBoard.getDevs(),
+          initialData: [],
+          child: Consumer<List<Dev>>(
+            builder: (_, data, __) => DevsList(
+              devs: data,
+              listType: ListType.CARD_FIRST,
```
Comment (Jansalvador1445) [inline lib/features/devboard/devboard_page.dart:41]: I think we don't need this one

File: lib/features/devboard/devs/devs_list.dart [hunk 2]
```diff
@@ -4,12 +4,16 @@ import 'package:flutter/material.dart';
 import 'package:font_awesome_flutter/font_awesome_flutter.dart';
 import 'package:url_launcher/url_launcher.dart';
 
+enum ListType { AVATAR, CARD_FIRST, CARD_SECOND, GRID }
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:7]: I think we dont need this

File: lib/features/devboard/devs/devs_list.dart [hunk 3]
```diff
@@ -4,12 +4,16 @@ import 'package:flutter/material.dart';
...
+
 class DevsList extends StatelessWidget {
   final List<Dev> devs;
+  final ListType listType;
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:11]: remove this one

File: lib/features/devboard/devs/devs_list.dart [hunk 4]
```diff
@@ -4,12 +4,16 @@ import 'package:flutter/material.dart';
...
 
   const DevsList({
     Key key,
     this.devs = const [],
+    this.listType = ListType.GRID,
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:16]: remove this one

File: lib/features/devboard/devs/devs_list.dart [hunk 5]
```diff
@@ -22,6 +26,45 @@ class DevsList extends StatelessWidget {
     /// Sort dev's ascending
     devs.sort((a, b) => a.name.compareTo(b.name));
 
+    if (this.listType == ListType.AVATAR) {
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:29]: remove this one

File: lib/features/devboard/devs/devs_list.dart [hunk 6]
```diff
@@ -22,6 +26,45 @@ class DevsList extends StatelessWidget {
...
+          scrollDirection: Axis.horizontal,
+          itemCount: devs.length,
+          itemBuilder: (_, i) => _DevListItem(
+            dev: devs[i],
+          ),
+        ),
+      );
+    }
+
+    if (this.listType == ListType.CARD_FIRST) {
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:42]: let's remove conditional statement, will use `Card_first`

File: lib/features/devboard/devs/devs_list.dart [hunk 7]
```diff
@@ -22,6 +26,45 @@ class DevsList extends StatelessWidget {
...
+          scrollDirection: Axis.horizontal,
+          itemCount: devs.length,
+          itemBuilder: (_, i) => _DevListCardFirst(
+            dev: devs[i],
+          ),
+        ),
+      );
+    }
+
+    if (this.listType == ListType.CARD_SECOND) {
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:55]: remove this one

File: lib/features/devboard/devs/devs_list.dart [hunk 8]
```diff
@@ -36,6 +79,373 @@ class DevsList extends StatelessWidget {
   }
 }
 
+class _DevListCardSecond extends StatelessWidget {
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:82]: remove this one

File: lib/features/devboard/devs/devs_list.dart [hunk 9]
```diff
@@ -36,6 +79,373 @@ class DevsList extends StatelessWidget {
...
+              ],
+            ),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class _DevListCardFirst extends StatelessWidget {
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:274]: From `_DevListCardFirst` to `_DevListCard`

File: lib/features/devboard/devs/devs_list.dart [hunk 10]
```diff
@@ -36,6 +79,373 @@ class DevsList extends StatelessWidget {
...
+  final Dev dev;
+
+  const _DevListCardFirst({Key key, this.dev}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final bool isDark = false;
+
+    final Color lightColor = Colors.white.withOpacity(.6);
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:283]: Let's add this to the `Constant/Colors` file or folder.

File: lib/features/devboard/devs/devs_list.dart [hunk 11]
```diff
@@ -36,6 +79,373 @@ class DevsList extends StatelessWidget {
...
+    final Color darkColor = Colors.black.withOpacity(.6);
+
+    final TextStyle title = TextStyle(
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:286]: I think we could embed this to the destionation

File: lib/features/devboard/devs/devs_list.dart [hunk 12]
```diff
@@ -36,6 +79,373 @@ class DevsList extends StatelessWidget {
...
+      fontSize: 18,
+      fontWeight: FontWeight.bold,
+      color: isDark ? Colors.white : Colors.black,
+    );
+    final TextStyle subTitle = TextStyle(
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:291]: same here

File: lib/features/devboard/devs/devs_list.dart [hunk 13]
```diff
@@ -36,6 +79,373 @@ class DevsList extends StatelessWidget {
...
+      fontSize: 13,
+      fontWeight: FontWeight.w300,
+      color: isDark ? lightColor : darkColor,
+    );
+    final TextStyle content = subTitle.copyWith(
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:296]: same here

File: lib/features/devboard/devs/devs_list.dart [hunk 14]
```diff
@@ -36,6 +79,373 @@ class DevsList extends StatelessWidget {
...
+      fontWeight: FontWeight.w200,
+      color: isDark ? lightColor : darkColor,
+    );
+    final String roles = dev.roles.join(" • ");
+    final Color iconColor = isDark ? Colors.blueAccent[100] : Colors.blueAccent;
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:301]: We can add this to the `Constant/Colors` file/folder

File: lib/features/devboard/devs/devs_list.dart [hunk 15]
```diff
@@ -36,6 +79,373 @@ class DevsList extends StatelessWidget {
...
+                Row(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  children: [
+                    CircleAvatar(
+                      backgroundImage: NetworkImage(
+                          '${StringConstant.baseImageURL}${dev.username}'),
+                      radius: 50,
+                    ),
+                    SizedBox(
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:409]: Let's use `Container` with `margin`

Comment (Jansalvador1445) [reply lib/features/devboard/devs/devs_list.dart:409]: Bottom and top

File: lib/features/devboard/devs/devs_list.dart [hunk 16]
```diff
@@ -36,6 +79,373 @@ class DevsList extends StatelessWidget {
...
+                      width: 120, // Tried expanded, but somehow doesn't work...
+                    ),
+                    _socials(dev.socials),
+                  ],
+                ),
+                SizedBox(
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:415]: remove this once container is added

File: lib/features/devboard/devs/devs_list.dart [hunk 17]
```diff
@@ -104,45 +514,64 @@ class _DevListItem extends StatelessWidget {
...
+          children: [
+            Container(
+              height: 220,
+              width: 220,
+              decoration: BoxDecoration(
+                shape: BoxShape.circle,
+                image: DecorationImage(
+                    image: NetworkImage(
+                        "${StringConstant.baseImageURL}${dev.username}"),
+                    fit: BoxFit.fill),
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:535]: `fit: BoxFit.fill,` -> Add comma only

File: lib/features/devboard/devboard_page.dart [hunk 18]
```diff
@@ -4,6 +4,8 @@ import 'package:devs/features/devboard/devs/devs_list.dart';
 import 'package:flutter/material.dart';
 import 'package:provider/provider.dart';
```
Comment (Jansalvador1445) [inline lib/features/devboard/devboard_page.dart:6]: Remove space

File: lib/features/devboard/devs/devs_list.dart [hunk 19]
```diff
@@ -36,6 +79,373 @@ class DevsList extends StatelessWidget {
...
+              ],
+            ),
+          )
+        ],
+      ),
+    );
+  }
+}
+
 class _DevListItem extends StatelessWidget {
```
Comment (Jansalvador1445) [inline lib/features/devboard/devs/devs_list.dart:449]: Remove this, since we will use the Second card

Comment (Jansalvador1445) [reply lib/features/devboard/devs/devs_list.dart:449]: No need for this will be removed.

File: lib/features/devboard/devs/devs_list.dart (hunk 11 above)
Comment (jose-bamboo) [reply lib/features/devboard/devs/devs_list.dart:286]: Could you clarify on this, where can I find the destination file/folder?

File: lib/features/devboard/devs/devs_list.dart (hunk 15 above)
Comment (jose-bamboo) [reply lib/features/devboard/devs/devs_list.dart:409]: Should I wrap the Column widget with margin or replace the SizedBox widget with a Container with margin?

File: lib/features/devboard/devs/devs_list.dart (hunk 11 above)
Comment (Jansalvador1445) [reply lib/features/devboard/devs/devs_list.dart:286]: `lib/constant/colors.dart` -> please add `colors.dart`

```
class ColorConstant {
  //Some Codes
}
```

**Comment by Jansalvador1445** [comment]
Can we do this(vertical position but same design/ui)? What do you think?

image

**Comment by jose-bamboo** [comment]
> Can we do this(vertical position but same design/ui)? What do you think?
> 
> image

Sure, I'll try it out right now.

**Comment by Jansalvador1445** [comment]
When I get home, i'll try to push on your branch.

**Comment by jose-bamboo** [comment]
image

image


I think it\s working. I made use of the other card I made awhile ago for mobile because the other card causes some of the widgets to overflow. Will push this in my branch now.

**Comment by Jansalvador1445** [comment]
Nice! Will check it now.

**Comment by Jansalvador1445** [comment]
Just pushed on your branch, can you `git pull` then test it out? If there's an error on it. thanks

**Comment by Jansalvador1445** [comment]
If everything is good, we can merge this one.

**Comment by jose-bamboo** [comment]
Got it, no errors encountered, though I experienced some widget overflowing but was able to fix it in my latest push.

**Comment by Jansalvador1445** [comment]
This is good on my side. Merging this now.

**Comment by Jansalvador1445** [comment]
Congrats @jose-bamboo! Wohooo! :D

**Comment by jose-bamboo** [comment]
> Congrats @jose-bamboo! Wohooo! :D

Really late reply, but thank you so much for the help! :)

**Comment by joshuadeguzman** [comment]
@allcontributors please add @jose-bamboo  for design, code

**Comment by joshuadeguzman** [comment]
@allcontributors please add @Jansalvador1445  for mentoring, code, bug

**Pull request merged**